lib/test_prof/before_all.rb

```diff
@@ -16,26 +16,28 @@ class << self
       attr_accessor :adapter
 
       def begin_transaction
+        raise "begin_transaction does not support a block" if block_given?
         raise AdapterMissing if adapter.nil?
 
         config.run_hooks(:begin) do
           adapter.begin_transaction
         end
-        yield
-      end
-
-      def within_transaction
-        yield
       end
 
       def rollback_transaction
-        raise AdapterMissing if adapter.nil?
```

Comment:
Chances of rollback method to be called with `adapter.nil?` are nearing zero, considering there's already such a check in `begin_transaction`.